packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -257,323 +231,412 @@ describe("Garbage Collection Tests", () => {
+                // Validate that all events are logged as expected.
+                garbageCollector.nodeUpdated(nodes[1], "Changed", Date.now(), testPkgPath);
+                garbageCollector.nodeUpdated(nodes[2], "Changed", Date.now(), testPkgPath);
+                garbageCollector.nodeUpdated(nodes[3], "Loaded", Date.now(), testPkgPath);
+                await garbageCollector.collectGarbage({});
+                assert(
+                    mockLogger.matchEvents([
+                        { eventName: changedEventName, timeout, id: nodes[1], pkg: eventPkg },
+                        { eventName: changedEventName, timeout, id: nodes[2], pkg: eventPkg },
+                        { eventName: loadedEventName, timeout, id: nodes[3], pkg: eventPkg },
+                    ]),
+                    "all events not generated as expected",
+                );
+            });
+        };
 
-            // Remove node 3's reference from node 2 so that it is still unreferenced. The GC details from the base
-            // summary is not loaded until the first time GC is run, so run GC.
-            defaultGCData.gcNodes[nodes[2]] = [];
-            await garbageCollector.collectGarbage({ runGC: true });
+        describe("Inactive events", () => {
+            const inactiveTimeoutMs = 500;
 
-            // Validate that changed and loaded events are logged for node 3 since it is inactive.
-            garbageCollector.nodeUpdated(nodes[3], "Changed", Date.now(), testPkgPath);
-            garbageCollector.nodeUpdated(nodes[3], "Loaded", Date.now(), testPkgPath);
-            await garbageCollector.collectGarbage({});
-            assert(
-                mockLogger.matchEvents([
-                    { eventName: changedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                    { eventName: loadedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg },
-                ]),
-                "inactive events not logged as expected",
-            );
+            beforeEach(() => {
+                injectedSettings["Fluid.GarbageCollection.TestOverride.InactiveTimeoutMs"] = inactiveTimeoutMs;
+            });
 
-            // Add reference from node 2 to node 3 and validate that we get revived event.
-            garbageCollector.addedOutboundReference(nodes[2], nodes[3]);
-            await garbageCollector.collectGarbage({});
-            assert(
-                mockLogger.matchEvents([
-                    { eventName: revivedEvent, timeout: inactiveTimeoutMs, id: nodes[3], pkg: eventPkg, fromId: nodes[2] },
-                ]),
-                "revived event not logged as expected",
+            tests(
+                inactiveTimeoutMs,
+                "GarbageCollector:InactiveObject_Revived",
+                "GarbageCollector:InactiveObject_Changed",
+                "GarbageCollector:InactiveObject_Loaded",
             );
```

Comment:
Why do you need to run these w/o the delete event and then again with the same info + delete event?  I.e. why isn't the "Sweep ready events" describe block sufficient to get coverage for everything this "Inactive events" block does?

Reply:
Ok I figured it out. The difference between the blocks is the relative length of the timeouts...?  Like for this describe block, inactiveObjectTimeoutMs is 500 which is used as the timeout to advance, and the delete event is missing so it skips validation of that stuff (since it won't be true).

Then for the other describe block, inactiveTimeout is the default 7 days (right?), and total sweep timeout is 31d+500ms, which is greater than 7 days, so both are hit.

I think there's probably a cleaner way to parametrize the shared test code but don't worry about it, they pass :)

Reply:
Yes, that is correct. We can validated both in the same tests too. I did this to make sure we can enable both independently as well.
Yes, I agree it can be cleaner. Please feel free to make a suggestion if you have any :-).